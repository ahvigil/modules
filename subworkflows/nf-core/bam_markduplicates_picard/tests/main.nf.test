nextflow_workflow {

    name "Test Workflow BAM_MARKDUPLICATES_PICARD"
    script "../main.nf"
    workflow "BAM_MARKDUPLICATES_PICARD"

    tag "picard"
    tag "picard/markduplicates"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "bam_markduplicates_picard"
    tag "subworkflows/bam_markduplicates_picard"
    tag "subworkflows/bam_stats_samtools"
    tag "bam_stats_samtools"
    tag "samtools"
    tag "samtools/flagstat"
    tag "samtools/idxstats"
    tag "samtools/index"
    tag "samtools/stats"
    
    test("homo_sapiens - bam") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                ]
                input[1] = [ 
                    [ id:'genome' ],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                ]
                input[2] = [ 
                    [ id:'genome' ],
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(path(workflow.out.bam.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.bai.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.flagstat.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.idxstats.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.stats.get(0).get(1))).match() },
                { assert path(workflow.out.metrics.get(0).get(1)).getText().contains("97") }
            )
        }
    }

    test("homo_sapiens - cram") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true)
                ]
                input[1] = [ [ id:'genome' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ]
                input[2]   = [ 
                    [ id:'genome' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(path(workflow.out.bam.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.bai.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.flagstat.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.idxstats.get(0).get(1))).match() },
                { assert snapshot(path(workflow.out.stats.get(0).get(1))).match() },
                { assert path(workflow.out.metrics.get(0).get(1)).getText().contains("0.999986") }
            )
        }
    }

}
