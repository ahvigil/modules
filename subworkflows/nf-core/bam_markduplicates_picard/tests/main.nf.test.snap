{
    "homo_sapiens - cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,6641dc05efa8384a061f378d86d922cd"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MarkDuplicates.metrics.txt:md5,e3b07a0fb99c8d50dfee6e2808202cdd"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.bai:md5,c41c60d8a94adebe53b6df80b6e90d38"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,1eb0987d2c8682c359977e7195527b4d"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,93b0ef463df947ede1f42ff60396c34d"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15"
                    ]
                ],
                "7": [
                    "versions.yml:md5,966dcea920866a87b55e665563864fc9",
                    "versions.yml:md5,99714fc707f125e4b1b5147e95fab2fe",
                    "versions.yml:md5,ad4b296f3d037cabb77941ede43bf9c4",
                    "versions.yml:md5,ccb22704a7d3adaba876f029b8083211",
                    "versions.yml:md5,dbb6a2e8c93fb4d082703191b57848cb"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.bai:md5,c41c60d8a94adebe53b6df80b6e90d38"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,6641dc05efa8384a061f378d86d922cd"
                    ]
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,93b0ef463df947ede1f42ff60396c34d"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15"
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MarkDuplicates.metrics.txt:md5,e3b07a0fb99c8d50dfee6e2808202cdd"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,1eb0987d2c8682c359977e7195527b4d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,966dcea920866a87b55e665563864fc9",
                    "versions.yml:md5,99714fc707f125e4b1b5147e95fab2fe",
                    "versions.yml:md5,ad4b296f3d037cabb77941ede43bf9c4",
                    "versions.yml:md5,ccb22704a7d3adaba876f029b8083211",
                    "versions.yml:md5,dbb6a2e8c93fb4d082703191b57848cb"
                ]
            }
        ],
        "timestamp": "2023-11-29T10:55:18.470325"
    },
    "homo_sapiens - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,3091fe6ba1b7530f382fe40b9fd8f45b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MarkDuplicates.metrics.txt:md5,37e8e2d51b6c0330289ba250b4db77e9"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.bai:md5,4d3ae8d013444b55e17aa0149a2ab404"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,f3f0e5aad236aae678ac5361b529a664"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,4f7ffd1e6a5e85524d443209ac97d783"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,df60a8c8d6621100d05178c93fb053a2"
                    ]
                ],
                "7": [
                    "versions.yml:md5,966dcea920866a87b55e665563864fc9",
                    "versions.yml:md5,99714fc707f125e4b1b5147e95fab2fe",
                    "versions.yml:md5,ad4b296f3d037cabb77941ede43bf9c4",
                    "versions.yml:md5,ccb22704a7d3adaba876f029b8083211",
                    "versions.yml:md5,dbb6a2e8c93fb4d082703191b57848cb"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.bai:md5,4d3ae8d013444b55e17aa0149a2ab404"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,3091fe6ba1b7530f382fe40b9fd8f45b"
                    ]
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,4f7ffd1e6a5e85524d443209ac97d783"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,df60a8c8d6621100d05178c93fb053a2"
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MarkDuplicates.metrics.txt:md5,37e8e2d51b6c0330289ba250b4db77e9"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,f3f0e5aad236aae678ac5361b529a664"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,966dcea920866a87b55e665563864fc9",
                    "versions.yml:md5,99714fc707f125e4b1b5147e95fab2fe",
                    "versions.yml:md5,ad4b296f3d037cabb77941ede43bf9c4",
                    "versions.yml:md5,ccb22704a7d3adaba876f029b8083211",
                    "versions.yml:md5,dbb6a2e8c93fb4d082703191b57848cb"
                ]
            }
        ],
        "timestamp": "2023-11-29T10:54:07.846026"
    }
}